---
title: "pipeline"
output: html_document
---

# For Nakazawaea holstii

reads <- "/Users/ocoelho00/Documents/biol4315/repos/yeast_project/Nakazawaea_holstii.fastq"

# Un-trimmed

## Run sequali on un-trimmed reads
```{bash, eval=FALSE}
sequali --outdir ./Un-trimmed/sequali_reports_holstii ./Un-trimmed/Nakazawaea_holstii.fastq
```

## Run flye on un-trimmed reads
```{bash, eval=FALSE}
flye --nano-raw ./Un-trimmed/Nakazawaea_holstii.fastq --out-dir ./Un-trimmed/flye_output_holstii --threads 8 --genome-size 13m
```
Assembly statistics:
    Total length:   13055090
    Fragments:      307
    Fragments N50:  69382
    Largest frg:    406130
    Scaffolds:      0
    Mean coverage:  8

___________________________________________________________________________________________________________________________________
___________________________________________________________________________________________________________________________________
___________________________________________________________________________________________________________________________________
___________________________________________________________________________________________________________________________________


# Trimmed reads

## Get trimmed fastq
```{bash, eval=FALSE}
filtlong --min_length 120 ./Un-trimmed/Nakazawaea_holstii.fastq > ./Trimmed/trimmed_Nakazawaea_holstii.fastq
```

```{bash, eval=FALSE}
sequali --outdir ./Trimmed/sequali_reports_holstii ./Trimmed/trimmed_Nakazawaea_holstii.fastq
```

## Run flye on trimmed reads
```{bash, eval=FALSE}
flye --nano-raw ./Trimmed/trimmed_Nakazawaea_holstii.fastq --out-dir ./Trimmed/flye_output_holstii --threads 8 --genome-size 13m
```
Assembly statistics:
    Total length:   13074047
    Fragments:      31
    Fragments N50:  70378
    Largest frg:    406126
    Scaffolds:      0
    Mean coverage:  8


# Trimmed funannotate

## Create mask 
```{bash, eval=FALSE}
docker run --rm -v "$PWD":/data -w /data --platform linux/amd64 nextgenusfs/funannotate funannotate mask -i ./Trimmed/flye_output_holstii/assembly.fasta -o ./Trimmed/trimmed_masked_assembly_holstii.fasta
```

## Run prediction
```{bash, eval=FALSE}
docker run --rm -v "$PWD":/data -w /data --platform linux/amd64 nextgenusfs/funannotate funannotate predict -i ./Trimmed/trimmed_masked_assembly_holstii.fasta -o ./Trimmed/prediction_output_holstii -s 'Nakazawaea holstii'
```

## Run annotation
```{bash, eval=FALSE}
docker run --rm -v "$PWD":/data -w /data --platform linux/amd64 nextgenusfs/funannotate funannotate annotate -i ./Trimmed/prediction_output_holstii/ -o ./Trimmed/annotate_output_holstii
```

## Run BUSCO
```{bash, eval=FALSE}
docker run --rm \
  -v "$PWD":/data \
  -w /data \
  ezlabgva/busco:v6.0.0_cv1 \
  busco -i ./Trimmed/flye_output_holstii/assembly.fasta -o ./Trimmed/busco_output_holstii -l saccharomycetaceae_odb12 -m genome -c 8
```

|C:48.1%[S:47.3%,D:0.8%],F:1.2%,M:50.6%,n:3282,E:48.6%                                     
    |1579    Complete BUSCOs (C)    (of which 768 contain internal stop codons)             
    |1553    Complete and single-copy BUSCOs (S)                                            
    |26      Complete and duplicated BUSCOs (D)                                              
    |41      Fragmented BUSCOs (F)                                                            
    |1662    Missing BUSCOs (M)                                                             
    |3282    Total BUSCO groups searched  
    
## Run Quast  (https://www.ebi.ac.uk/ena/browser/view/GCA_003706265.1)
```{bash, eval=FALSE}
docker run --rm -v "$(pwd)":/data -w /data quay.io/biocontainers/quast:5.3.0--py313pl5321h5ca1c30_2 \
  quast.py ./Trimmed/flye_output_holstii/assembly.fasta \
  -r ./Un-trimmed/GCA_003706265.1_holstii.fasta \
  -o ./Trimmed/quast_output_holstii
```

ambiguity: one, min local misassembly length: 200, min extensive misassembly length: 1000
N50 = 70378, L50 = 42, auN = 120651.3, Total length = 13073587, GC % = 35.61, # N's per 100 kbp =  0.00
Largest alignment = 77998, NA50 = None, LA50 = None

___________________________________________________________________________________________________________________________________
___________________________________________________________________________________________________________________________________
___________________________________________________________________________________________________________________________________
___________________________________________________________________________________________________________________________________

# For Nakazawaea holstii

reads <- "/Users/ocoelho00/Documents/biol4315/repos/yeast_project/Nakazawaea_popuil.fastq"

# Un-trimmed

## Run sequali on un-trimmed reads
```{bash, eval=FALSE}
sequali --outdir ./Un-trimmed/sequali_reports_popuil ./Un-trimmed/Nakazawaea_popuil.fastq
```

# Trimmed reads

## Get trimmed fastq
```{bash, eval=FALSE}
filtlong --min_length 120 ./Un-trimmed/Nakazawaea_popuil.fastq > ./Trimmed/trimmed_Nakazawaea_popuil.fastq
```

```{bash, eval=FALSE}
sequali --outdir ./Trimmed/sequali_reports_popuil ./Trimmed/trimmed_Nakazawaea_popuil.fastq
```

## Run flye on trimmed reads
```{bash, eval=FALSE}
flye --nano-raw ./Trimmed/trimmed_Nakazawaea_popuil.fastq --out-dir ./Trimmed/flye_output_popuil --threads 8 --genome-size 13m
```
Assembly statistics:
    Total length:   13831975
    Fragments:      20
    Fragments N50:  1947967
    Largest frg:    2729774
    Scaffolds:      0
    Mean coverage:  46



# Trimmed funannotate

## Create mask 
```{bash, eval=FALSE}
docker run --rm -v "$PWD":/data -w /data --platform linux/amd64 nextgenusfs/funannotate funannotate mask -i ./Trimmed/flye_output_popuil/assembly.fasta -o ./Trimmed/trimmed_masked_assembly_popuil.fasta
```

## Run prediction
```{bash, eval=FALSE}
docker run --rm -v "$PWD":/data -w /data --platform linux/amd64 nextgenusfs/funannotate funannotate predict -i ./Trimmed/trimmed_masked_assembly_popuil.fasta -o ./Trimmed/prediction_output_popuil -s 'Nakazawaea popuil'
```

## Run annotation
```{bash, eval=FALSE}
docker run --rm -v "$PWD":/data -w /data --platform linux/amd64 nextgenusfs/funannotate funannotate annotate -i ./Trimmed/prediction_output_popuil/ -o ./Trimmed/annotate_output_popuil
```

## Run BUSCO
```{bash, eval=FALSE}
docker run --rm \
  -v "$PWD":/data \
  -w /data \
  ezlabgva/busco:v6.0.0_cv1 \
  busco -i ./Trimmed/flye_output_popuil/assembly.fasta -o ./Trimmed/busco_output_popuil -l saccharomycetaceae_odb12 -m genome -c 8
```

 |C:48.9%[S:48.2%,D:0.7%],F:1.3%,M:49.8%,n:3282,E:49.2%                                   
    |1606    Complete BUSCOs (C)    (of which 790 contain internal stop codons)             
    |1583    Complete and single-copy BUSCOs (S)                                          
    |23    Complete and duplicated BUSCOs (D)                                               
    |42    Fragmented BUSCOs (F)                                                            
    |1634    Missing BUSCOs (M)                                                            
    |3282    Total BUSCO groups searched  
    
## Run Quast  (https://www.ebi.ac.uk/ena/browser/view/GCA_030557235)
```{bash, eval=FALSE}
docker run --rm -v "$(pwd)":/data -w /data quay.io/biocontainers/quast:5.3.0--py313pl5321h5ca1c30_2 \
  quast.py ./Trimmed/flye_output_popuil/assembly.fasta \
  -r ./Un-trimmed/GCA_030557235.1_popuil.fasta \
  -o ./Trimmed/quast_output_popuil
```

ambiguity: one, min local misassembly length: 200, min extensive misassembly length: 1000
N50 = 1947967, L50 = 3, auN = 1777300.2, Total length = 13831975, GC % = 37.52, # N's per 100 kbp =  0.00
Largest alignment = 400630, NA50 = 18817, NGA50 = 21266, LA50 = 87, LGA50 = 80




holstii_eggnog <- "/Users/ocoelho00/Documents/biol4315/repos/yeast_project/eggnog/nakazawaea_holstii/MM_pu8bw7jh.emapper.annotations.xlsx"
populi_eggnog <- "/Users/ocoelho00/Documents/biol4315/repos/yeast_project/eggnog/nakazawaea_populi/MM_jx78ya44.emapper.annotations.xlsx"





=== Computational Environment ===
R version   : R version 4.4.0 (2024-04-24) 
Platform    : aarch64-apple-darwin20 
OS          : darwin20 
System      : Darwin 24.6.0 
Machine     : arm64 
Node name   : T450-C003-94656 
R home      : /Library/Frameworks/R.framework/Resources 
CPU cores   : 8 


=== Tool Versions ===
     tool                                                      path found         version
  sequali  /Users/ocoelho00/miniconda3/envs/sequali_env/bin/sequali  TRUE           1.0.1
 filtlong /Users/ocoelho00/miniconda3/envs/sequali_env/bin/filtlong  TRUE Filtlong v0.3.1
  chopper  /Users/ocoelho00/miniconda3/envs/sequali_env/bin/chopper  TRUE  chopper 0.11.0
     flye     /Users/ocoelho00/miniconda3/envs/sequali_env/bin/flye  TRUE     2.9.6-b1802
    quast /Users/ocoelho00/miniconda3/envs/sequali_env/bin/quast.py FALSE           5.3.0
    busco    /Users/ocoelho00/miniconda3/envs/sequali_env/bin/busco FALSE           6.0.0